### PR TITLE
PPTC GUI: Purge all versioned caches on click

### LINKED ISSUE #1446: PPTC: Purging doesn't delete all cache
Seems like when purging PPTC it only deletes cache of the latest update. Since Animal Crossing makes around 300 MB per cache it gets quite big if you don't purge it before updating the game. Think purging should purge all caches related to PPTC?
image

### PR DESCRIPTION
**Pull request by mageven**
Fixes #1446

### CODE REVIEW

File: Ryujinx/Ui/GameTableContextMenu.cs [hunk 1]
```diff
@@ -664,30 +665,36 @@ private void OpenPtcDir_Clicked(object sender, EventArgs args)
...
-                Text           = "You are about to delete the PPTC cache. Are you sure you want to proceed?",
+                Text           = $"You are about to delete the PPTC cache for '{tableEntry[0]}'. Are you sure you want to proceed?",
                 WindowPosition = WindowPosition.Center
             };
-            
-            if (warningDialog.Run() == (int)ResponseType.Yes)
+
+            var cacheFiles = new List<FileInfo>();
+
+            if(mainDir.Exists)   { cacheFiles.AddRange(mainDir.EnumerateFiles("*.cache")); }
```
Comment (AcK77) [inline Ryujinx/Ui/GameTableContextMenu.cs:683]:
```suggestion
            if (mainDir.Exists)   { cacheFiles.AddRange(mainDir.EnumerateFiles("*.cache")); }
```

File: Ryujinx/Ui/GameTableContextMenu.cs [hunk 2]
```diff
@@ -664,30 +665,36 @@ private void OpenPtcDir_Clicked(object sender, EventArgs args)
...
+            if(backupDir.Exists) { cacheFiles.AddRange(backupDir.EnumerateFiles("*.cache")); }
```
Comment (AcK77) [inline Ryujinx/Ui/GameTableContextMenu.cs:684]:
```suggestion
            if (backupDir.Exists) { cacheFiles.AddRange(backupDir.EnumerateFiles("*.cache")); }
```

File: Ryujinx/Ui/GameTableContextMenu.cs [hunk 3]
```diff
@@ -664,30 +665,36 @@ private void OpenPtcDir_Clicked(object sender, EventArgs args)
...
             {
-                if (File.Exists(mainPath))
+                foreach (FileInfo file in cacheFiles)
                 {
-                    File.Delete(mainPath);
-                }
-                if (File.Exists(backupPath))
-                {
-                    File.Delete(backupPath);
+                    try { file.Delete(); }
```
Comment (AcK77) [inline Ryujinx/Ui/GameTableContextMenu.cs:690]:
```suggestion
                    try 
                    { 
                        file.Delete(); 
                    }
```

File: Ryujinx/Ui/GameTableContextMenu.cs [hunk 4]
```diff
@@ -664,30 +665,39 @@ private void OpenPtcDir_Clicked(object sender, EventArgs args)
         
         private void PurgePtcCache_Clicked(object sender, EventArgs args)
         {
-            string titleId       = _gameTableStore.GetValue(_rowIter, 2).ToString().Split("\n")[1].ToLower();
-            string cacheFileName = _gameTableStore.GetValue(_rowIter, 4) + ".cache";
+            var tableEntry = _gameTableStore.GetValue(_rowIter, 2).ToString().Split("\n");
+            string titleId = tableEntry[1].ToLower();
```
Comment (jduncanator) [inline Ryujinx/Ui/GameTableContextMenu.cs:669]: If the string does not contain a new line, this will throw a IOR exception.

Comment (mageven) [reply Ryujinx/Ui/GameTableContextMenu.cs:669]: All of the code in that file assumes that there's a newline.

Comment (Xpl0itR) [reply Ryujinx/Ui/GameTableContextMenu.cs:669]: There will always be a newline, in the UI the 1st line is the title name and the second line is the title id

**Pull request merged**